aspnetcore/blazor/hosting-models.md

```diff
@@ -77,24 +77,77 @@ There are downsides to server-side hosting:
 
 &dagger;The *blazor.server.js* script is served from an embedded resource in the ASP.NET Core shared framework.
 
+### Comparison to server-rendered UI
+
+One way to understand Blazor Server apps is to understand how it differs from traditional models for rendering UI in ASP.NET Core apps using Razor views or Razor Pages. Both models use the Razor language to describe HTML content, but they significantly differ in how markup is rendered.
+
+When a Razor Page or view is rendered, every line of Razor code emits HTML in text form. After rendering, the server disposes of the page or view instance, including any state that was produced. When another request for the page occurs, for instance when server validation fails and the validation summary is displayed:
+
+* The entire page is rerendered to HTML text again.
+* The page is sent to the client.
+
+A Blazor app is composed of reusable elements of UI called *components*. A component contains C# code, markup, and other components. When a component is rendered, Blazor produces a graph of the included components similar to an HTML or XML Document Object Model (DOM). This graph includes component state held in properties and fields. Blazor evaluates the component graph to produce a binary representation of the markup. The binary format can be:
+
+* Turned into HTML text (during prerendering).
+* Used to efficiently diff the output at different times.
+
+<!--
+
+We probably should write out what "diff" means. How about ...
+
+* Used to efficiently find differences, usually called a *diff*, between the current component graph and a new component graph. A diff reflects changes in state at different times during rendering.
+
+... or what would you like it to say?
+
+-->
+
+A UI update in Blazor is triggered by:
+
+* User interaction, such as selecting a button.
+* App triggers, such as a timer.
+
+The graph is rerendered, and a UI diff is calculated. This diff is the smallest set of DOM edits required to update the UI on the client. The diff is sent to the client in a binary format and applied by the browser.
+
+A component is disposed after the user navigates away from it on the client. While a user is interacting with a component, the component's state (services, resources) must be held in the server's memory. Because the state of many components might be maintained by the server concurrently, resource exhaustion is a concern that must be addressed. For more information, see <xref:security/blazor/server-side>.
```

Comment:
>  resource exhaustion is a concern that must be addressed

Should we call out that it's memory exhaustion? You generally don't run out of CPU, network, or disk in most common cases.

Reply:
> For more information, see <xref:security/blazor/server-side>

Should this have a segue in to why the security stuff is useful. "We recommend looking at ... for some guidance on how to author your app to ensure the best use of server memory.